errors.go

```diff
@@ -199,6 +199,8 @@ var (
 
 	ErrRequiredName = errors.New("name is required")
 
+	ErrRequiredQuery = errors.New("invalid attribute\n\nQuery can't be blank")
```

Comment:
The error message doesn't feel like it fits in with the rest of them here.   Perhaps this?
```suggestion
	ErrRequiredQuery = errors.New("query can not be empty")
```